lib/ecto/schema.ex

```diff
@@ -200,7 +214,7 @@ defmodule Ecto.Schema do
       Module.register_attribute(__MODULE__, :ecto_assocs, accumulate: true)
       Module.register_attribute(__MODULE__, :ecto_raw, accumulate: true)
 
-      Module.put_attribute(__MODULE__, :struct_fields, {:__state__, :built})
+      Module.put_attribute(__MODULE__, :struct_fields, {:__meta__, %Metadata{state: :built}})
```

Comment:
Let's store the source in the metadata too so it is always available (instead of nil).